Checking reports

despiritosanto opened the pull request.

marcjones commented on features/step_definitions/request_licences_functionality_steps.rb:
```diff
@@ -0,0 +1,8 @@
+Given /^I navigate to Reports page$/ do
+  starting_page
+  on(ReportPage).proceed_to_report_tab
+  assert_that ('Reports are correctly displayed') {
```
I don't think it's a good idea having assertions in ```Given``` steps, the ```Then``` steps should be the ones with the test part in. If we need to assert this then put it in a ```Then``` step. If you are just waiting for the page to load then it's better to have a wait on expected condition in that page.

marcjones commented on features/support/env.rb:
```diff
@@ -22,13 +22,13 @@ def browser_name
   if env_variable == nil || env_variable == ""
     puts env_variable = "www"
-  elsif (env_variable.start_with? "integration") || (env_variable.start_with? "staging")
+  elsif (env_variable.include? "integration") || (env_variable.start_with? "staging")
```
Maybe you also want to change ```start_with``` to ```include``` for "staging" too, as I can predict "pro-staging" at some point

marcjones commented on features/support/pages/Report_Page.rb:
```diff
@@ -0,0 +1,49 @@
+is_tabs_correct&&is_panels_correct&is_Req_Lec_displayed
+end
+
+def verify_tabs_and_panels(is_tabs)
```
To quote Bob Martin...

"Functions should do one thing. They should do it well. They should do it only."

Separate methods for tabs and panels please. This one is too complex.

marcjones commented:
@despiritosanto couple of comments, will look again tomorrow

despiritosanto commented:
Cool @marcjones , 
Regarding to assert in 'Given', It just for test, coz I couldn't implement the rest of flow, but normally I onle use assertion at end of test 'Then'. I'll change it when I update the scenario.
In env.rb, I had changed it to start_with to include. And finaly in Report_page.rb, I agree with you, so I'll separate that function in 2 in order to make it easyer to understand.

I'll ping you when its ready, coz I'm going to add login scenario I implemented yesterday.

Thanks!

kumariswarnim commented on features/user/Registry/B2B_Invite_Students_Functionality.feature:
```diff
@@ -5,15 +5,12 @@ Feature: Invite students Functionality Feature
+#    When I Proceed to create a new classroom
+#    And I should see a invite option suggestion
+#    And I proceed to invite students
+#    Then I should see the new invited student displayed in classroom page
```
@despiritosanto Please modify the above scenarios as:
Scenario: Invite student to a new classroom
Given I Proceed to create a new classroom
And I see an invite option suggestion
When I proceed to invite students
Then I should see the new invited student displayed in classroom page

kumariswarnim commented on Properties/users_data.yml:
```diff
@@ -8,7 +8,7 @@ users:
-    email: 'b2btest3@mail.busuu.com'
+    email: 'deotest@harakirimail.com'
     <<: *busuu_pro_defaults_pass
 
```
Why are we using email address as 'b2btest3@mail.busuu.com'. Can you please use something like "qa+{generateRandomString}@busuu.com"

kumariswarnim commented on Properties/users_data.yml:
```diff
@@ -1,13 +1,19 @@
-    password: 'testing123'
+    email: 'b2btest2@mail.busuu.com'
+    <<: *busuu_pro_defaults_pass
 
```
Why are we using email address as 'b2btest3@mail.busuu.com'. Can you please use something like "qa+{generateRandomString}@busuu.com"

despiritosanto commented:
Cool @kumariswarnim ,

I already updated the feature file with "Given" instead of "when" in the first step, I've forgotten to updated it when I remove the "Background notification".

I also had changed the user data to use "b2bwebautomation+{generateRandomString}@busuu.com".

Thanks!

kumariswarnim commented on Properties/users_data.yml:
```diff
@@ -4,15 +4,15 @@ busuu_pro_default_password: &busuu_pro_defaults_pass
-    email: 'b2btest2@mail.busuu.com'
+    email: 'b2bwebautomation002@busuu.com'
     <<: *busuu_pro_defaults_pass
 
```
@despiritosanto Please don't add 'b2bwebautomation002@busuu.com' as this is not the valid email address. Please use "qa+{generateRandomString}@busuu.com" as this will redirect the email to QA's email addresses. Please don't forget to put "+"

kumariswarnim commented on features/user/Registry/B2B_Invite_Students_Functionality.feature:
```diff
@@ -6,11 +6,11 @@ Feature: Invite students Functionality Feature
+  Scenario: Invite student to a new classroom
+    Given I Proceed to create a new classroom
+    And I should see a invite option suggestion
+    And I proceed to invite students
```
@despiritosanto  please change 
 "And I proceed to invite students" with "When I proceed to invite student "

despiritosanto commented:
I update the user data file to use user like qa+{randomString}web@busuu.com

Thanks

despiritosanto commented:
Done, I also update "And I proceed to invite students" with "When I proceed to invite student "

kumariswarnim commented:
Thanks @despiritosanto.
